src/config.rs

```diff
@@ -0,0 +1,383 @@
+        if let Some(envs) = target_envs {
+            passthrough_target = Some(vec![]);
+            passthrough_target
+                .as_mut()
+                .map(|vec| vec.extend(envs.split_whitespace().map(|v| v.to_string())));
+        }
+        let build_envs = self.get_build_var("env_passthrough");
+        let mut passthrough_build = None;
+        if let Some(envs) = build_envs {
+            passthrough_build = Some(vec![]);
+            passthrough_build
+                .as_mut()
+                .map(|vec| vec.extend(envs.split_whitespace().map(|v| v.to_string())));
+        }
+
+        (passthrough_build, passthrough_target)
+    }
+
+    fn volumes(&self, target: &Target) -> (Option<Vec<String>>, Option<Vec<String>>) {
+        let target_envs = self.get_target_var(target, "ENV_VOLUMES");
+        let mut volumes_target = None;
+        if let Some(envs) = target_envs {
+            volumes_target = Some(vec![]);
+            volumes_target
+                .as_mut()
+                .map(|vec| vec.extend(envs.split_whitespace().map(|v| v.to_string())));
+        }
+        let build_envs = self.get_build_var("ENV_VOLUMES");
+        let mut volumes_build = None;
+        if let Some(envs) = build_envs {
+            volumes_build = Some(vec![]);
+            volumes_build
+                .as_mut()
+                .map(|vec| vec.extend(envs.split_whitespace().map(|v| v.to_string())));
+        }
+        // validate no =
+
+        (volumes_build, volumes_target)
+    }
+}
+
+#[derive(Debug)]
+pub struct Config {
+    toml: Option<Toml>,
+    env: Environment,
+}
+
+impl Config {
+    pub fn new(toml: Option<Toml>) -> Self {
+        Config {
+            toml,
+            env: Environment::new(),
+        }
+    }
+    
+    #[cfg(test)]
+    fn new_with(toml: Option<Toml>, env: Environment) -> Self {
+        Config {
+            toml,
+            env,
+        }
+    }
+    pub fn xargo(&self, target: &Target) -> Result<Option<bool>> {
+        let (build_xargo, target_xargo) = self.env.xargo(target)?;
+        let (toml_build_xargo, toml_target_xargo) = if let Some(ref toml) = self.toml {
+            toml.xargo(target)?
+        } else {
+            (None, None)
+        };
+
+        match (build_xargo, toml_build_xargo) {
+            (xargo @ Some(_), _) => return Ok(xargo),
+            (None, xargo @ Some(_)) => return Ok(xargo),
+            (None, None) => {}
+        };
+
+        match (target_xargo, toml_target_xargo) {
+            (xargo @ Some(_), _) => return Ok(xargo),
+            (None, xargo @ Some(_)) => return Ok(xargo),
+            (None, None) => {}
+        };
+        Ok(None)
+    }
+    pub fn image(&self, target: &Target) -> Result<Option<String>> {
+        let env_value = self.env.image(target);
+        if let Some(env_value) = env_value {
+            return Ok(Some(env_value));
+        }
+        self.toml.as_ref().map_or(Ok(None), |t| t.image(target))
+    }
```

Comment:
Add whitespace between functions.

```suggestion
    }
    
```

Reply:
I reformatted